tests/units/typing/tensor/test_torch_tensor.py

```diff
@@ -152,3 +152,17 @@ def test_parametrized_operations():
     assert isinstance(t_result, torch.Tensor)
     assert isinstance(t_result, TorchTensor)
     assert isinstance(t_result, TorchTensor[128])
+
+
+def test_deepcopy():
+    from docarray import BaseDocument
+
+    class MMdoc(BaseDocument):
+        embedding: TorchEmbedding
+
+    doc = MMdoc(embedding=torch.randn(32))
+    doc_copy = doc.copy(deep=True)
+
+    assert id(doc) != id(doc_copy)
+    assert id(doc.embedding) != id(doc_copy.embedding)
```

Comment:
I think that this test is not enough.

Two pytorch variable can have different ID, which mean they are different variable, but still point to the same underlying data storage. 

You can use `tensor.data_ptr()` to get the pytorch pointer to the actual data (we use it in a couple of otehr test).


Another thing that you can do is to actually check that if you changed the value of `doc_copy` it does not affect `doc`